datasets/sogou_news/sogou_news.py

```diff
@@ -0,0 +1,99 @@
+# coding=utf-8
+# Copyright 2020 The TensorFlow Datasets Authors and the HuggingFace NLP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Sogou News"""
+
+from __future__ import absolute_import, division, print_function
+import os
+
+import nlp
+
+import sys
+import csv
+
+csv.field_size_limit(sys.maxsize)
+
+
+
+_CITATION = """\
+@misc{zhang2015characterlevel,
+    title={Character-level Convolutional Networks for Text Classification},
+    author={Xiang Zhang and Junbo Zhao and Yann LeCun},
+    year={2015},
+    eprint={1509.01626},
+    archivePrefix={arXiv},
+    primaryClass={cs.LG}
+}
+"""
+
+_DESCRIPTION = """\
+The Sogou News dataset is a mixture of 2,909,551 news articles from the SogouCA and SogouCS news corpora, in 5 categories. 
+The number of training samples selected for each class is 90,000 and testing 12,000. Note that the Chinese characters have been converted to Pinyin.
+classification labels of the news are determined by their domain names in the URL. For example, the news with
+URL http://sports.sohu.com is categorized as a sport class.
+"""
+
+_DATA_URL = "https://s3.amazonaws.com/fast-ai-nlp/sogou_news_csv.tgz"
+
+
+class Sogou_News(nlp.GeneratorBasedBuilder):
+    """Sogou News dataset"""
+
+    def _info(self):
+        return nlp.DatasetInfo(
+            description=_DESCRIPTION,
+            features=nlp.Features(
+                {
+                    "title": nlp.Value("string"),
+                    "content": nlp.Value("string"),
+                    "class": nlp.Value("string")
```

Comment:
Maybe we can use `ClassLabel` ?
Feel free to hardcode the classes instead of reading `classes.txt` to define the labels.